connector/src/test/scala/com/basho/riak/spark/rdd/timeseries/TimeSeriesReadTest.scala

```diff
@@ -77,6 +86,55 @@ class TimeSeriesReadTest extends AbstractTimeSeriesTest with AbstractRDDTest {
         |]
       """.stripMargin, stringify(data))
   }
+
+  @Test
+  def riakTSRDDToDataFrameConvertTimestamp(): Unit = {
+    val sqlContext = new org.apache.spark.sql.SQLContext(sc)
+    import sqlContext.implicits._
+
+    val structType = StructType(List(
+      StructField(name = "time", dataType = LongType),
+      StructField(name = "user_id", dataType = StringType),
+      StructField(name = "temperature_k", dataType = DoubleType))
+    )
+
+    val df = sc.riakTSBucket[org.apache.spark.sql.Row](bucketName, schema = Some(structType))
+      .sql(s"SELECT time, user_id, temperature_k FROM $bucketName $sqlWhereClause")
+      .map(r => TimeSeriesData(r.getLong(0), r.getString(1), r.getDouble(2)))
+      .toDF()
+
+    df.registerTempTable("test")
+
+    val data = sqlContext.sql("select * from test").toJSON.collect()
+
+    // -- verification
+    assertEqualsUsingJSONIgnoreOrder(
+      """
+        |[
+        |   {time:111111, user_id:'bryce', temperature_k:305.37},
+        |   {time:111222, user_id:'bryce', temperature_k:300.12},
+        |   {time:111333, user_id:'bryce', temperature_k:295.95},
+        |   {time:111444, user_id:'ratman', temperature_k:362.121},
+        |   {time:111555, user_id:'ratman', temperature_k:3502.212}
+        |]
+      """.stripMargin, stringify(data))
+  }
+
+  @Test
+  def sqlReadWithIncorrectSchemaShouldFail(): Unit = {
+    expectedException.expect(classOf[SparkException])
+    expectedException.expectMessage("Provided schema does not match the riak row columns")
```

Comment:
We need to specify what exact column is wrong as well as what is wrong: name and/or type. Feel free to create a separate issue for that.